cns/restserver/api.go

```diff
@@ -1119,14 +1140,24 @@ func (service *HTTPRestService) publishNetworkContainer(w http.ResponseWriter, r
 	}
 
 	logger.Request(service.Name, &reqCopy, err)
+
+	// TODO - refactor this method for better error handling
 	if err != nil {
 		return
 	}
 
+	var associatedInterfaceID, authToken string
+	associatedInterfaceID, authToken, err = getAuthTokenAndInterfaceIdFromNcURL(creteNcUrlCopy)
+	if err != nil {
+		logger.Errorf("[Azure-CNS] %+v", err)
+		w.WriteHeader(http.StatusBadRequest)
```

Comment:
- more context on log line would helpful, like the action being taken at the moment of error (publish network container) and things that may not be in the error string (like nc id)
- is there no payload returned to client? i think this change merits a response body with error details, since otherwise all we will see in DNC logs is 400

Reply:
Added more context and returned a cns response that DNC will parse